### ISSUE TITLE
confusing error message when looping analyses

### ISSUE THREAD

**Issue by andrewljohnson** [tflearn/tflearn#195]
I am training a shallow network for each US state. The first analysis works, the second throws the following error, when [model.fit is called in this code](https://github.com/trailbehind/DeepOSM/blob/feature/analysis-cycle/src/single_layer_network.py#L74).

It's a bit confusing that the error message relates to tensorflow instead of tflearn. Maybe tflearn should wrap _with tf.Graph().as_default()_ 

    File "/DeepOSM/src/single_layer_network.py", line 75, in train_with_data
    run_id='mlp')
    File "/usr/local/lib/python2.7/dist-packages/tflearn/models/dnn.py", line 153, in fit
    self.targets)
    File "/usr/local/lib/python2.7/dist-packages/tflearn/utils.py", line 253, in feed_dict_builder
    ". If you are using notebooks, please " 
    Exception: 274 inputs feeded, but expected: 2. If you are using notebooks, please make sure that you didn't run graph construction cell multiple time, or try to enclose your graph within `with tf.Graph().as_default():`

**Comment by aymericdamien** [comment]
Make sure that you are only building your graph once, you can encapsulate your code with 'tf.Graph().as_default()' if you need to build new graph for each loop.

**Comment by andrewljohnson** [comment]
I figured it out, but found the error message to be misleading, and hard to Google.

**Comment by aymericdamien** [comment]
True! I will look into that, maybe we should change it, or try to get more details about what kind of error is occuring.

**Comment by mbaniasad** [comment]
I also got do work around it with the normal lstm sample that was in the tutorial but with dynamic  enabled I get this error
```
import tensorflow as tf
with tf.Graph().as_default():
    train, test, _ = imdb.load_data(path=dataset_path, n_words=10000,
                                    valid_portion=0.1)
    trainX, trainY = train
    testX, testY = test

    # Data preprocessing
    # NOTE: Padding is required for dimension consistency. This will pad sequences
    # with 0 at the end, until it reaches the max sequence length. 0 is used as a
    # masking value by dynamic RNNs in TFLearn; a sequence length will be
    # retrieved by counting non zero elements in a sequence. Then dynamic RNN step
    # computation is performed according to that length.
    trainX = pad_sequences(trainX, maxlen=100, value=0.)
    testX = pad_sequences(testX, maxlen=100, value=0.)
    # Converting labels to binary vectors
    trainY = to_categorical(trainY, nb_classes=2)
    testY = to_categorical(testY, nb_classes=2)

    # Network building
    net = tflearn.input_data([None, 100])
    # Masking is not required for embedding, sequence length is computed prior to
    # the embedding op and assigned as 'seq_length' attribute to the returned Tensor.
    net = tflearn.embedding(net, input_dim=10000, output_dim=128)
    net = tflearn.lstm(net, 128, dropout=0.8, dynamic=True)
    net = tflearn.fully_connected(net, 2, activation='softmax')
    net = tflearn.regression(net, optimizer='adam', learning_rate=0.001,
                             loss='categorical_crossentropy')

    # Training
    model = tflearn.DNN(net, tensorboard_verbose=0, tensorboard_dir='./tflearn_logs/')
    model.fit(trainX, trainY, validation_set=(testX, testY), show_metric=True,
              batch_size=32, n_epoch=1)
```

**Comment by aymericdamien** [comment]
I have never seen such error. I may be related to TensorFlow rather than TFLearn.

**Comment by kevinfenger** [comment]
This is old, but since I ran into this issue, and have a fix/workaround. 

Due to this line. 

`net = tflearn.lstm(net, 128, dropout=0.8, dynamic=True)`

I haven't looked into exactly why, but my guess is the dynamic loading loads some operation in global scope, and causes the assign mess. 

Remove the dynamic=True, and the model can load/predict fine.

**Comment by seinberg** [comment]
Hi, I'm seeing the same issue when trying to do dynamic LSTM sequence lengths. I get this error:
`Operation 'LSTM/LSTM/cond/Assign' has been marked as not fetchable.`

This happens both in my custom implementation using TFLearn on sequences of ~350 as well as with the dynamic LSTM python file included in the NLP directory of the github repo.

I'm running TensorFlow / TFLearn from Python 3.6 - is that possibly a reason why?

**Comment by seinberg** [comment]
Hmm, indeed, when I do nothing except change the version of Python to version 3.5 the TFLearn examples train successfully.

**Comment by seinberg** [comment]
One last note - the only real difference between my code and the example TFLearn code is that I'm using pre-trained word embeddings by setting trainable=False and then getting the layer/variable by name and setting the weights. When not in dynamic mode, the LSTM trains just fine, but in dynamic mode it breaks with the fetch mode error.

**Comment by anassalamah** [comment]
Did you find a solution to 'LSTM/LSTM/cond/Assign' has been marked as not fetchable.' when turning dynamic to True and using pre-trained word embeddings?

**Comment by CWOA** [comment]
Has anyone found a solution to this (i.e. "Operation u'LSTM/LSTM/cond/Assign' has been marked as not fetchable."). I'm getting the same issue when defining a LSTM cell with the `dynamic=True` argument.

**Comment by seinberg** [comment]
Yeah - use keras instead :)
